src/main/scala/com/ubirch/webui/services/ApiDevices.scala

```diff
@@ -552,13 +552,40 @@ class ApiDevices(implicit val swagger: Swagger)
       val nowUtcMillis = System.currentTimeMillis()
       implicit val realmName: String = userInfo.realmName
 
-      val user = UserFactory.getByUsername(userInfo.userName)
-
       val res = GraphOperations.bulkGetUpps(user, hwDevicesIdString, beginningDayUtcMillis, nowUtcMillis)
       Ok(uppsToJson(res))
     }
   }
 
+  val getLastHash: SwaggerSupportSyntax.OperationBuilder =
+    (apiOperation[LastHash]("getLastHashDevice")
+      summary "Get the last hash produced by a device"
+      description "Get the last hash produced by a device"
+      tags "Devices"
+      parameters (
+      swaggerTokenAsHeader,
```

Comment:
doesn't need a description?

Reply:
True, it's quite minimalistic, will change that